jekyll/_cci2/docker-layer-caching.md

```diff
@@ -1,293 +1,124 @@
 
-If we were to change the first step in our example Dockerfile (for example, perhaps we want to pull from a different Elixir base image), then our entire cache for this image would be invalidated even if every other part of our Dockerfile stayed the same.
+If part of the Dockerfile changes (which changes part of the image), a subsequent run of the exact same job with the modified Dockerfile may still finish faster than rebuilding the entire image. This is because the cache can still be used for the first few steps that did not change in the Dockerfile. The steps that follow the change must be rerun because the Dockerfile change invalidates the cache for those layers.
 
-## Video: overview of Docker layer caching
-{: #video-overview-of-docker-layer-caching }
+If you change something in your Dockerfile, all of the later steps (from the point that the change was made) are invalidated and the layers have to be rebuilt. When some of the steps remain the same (the steps before the one you removed), those steps can be reused. So, it is still faster than rebuilding the entire image.
 
-In the video example, the job runs all of the steps in a Dockerfile with the `docker_layer_caching: true` for the `setup_remote_docker` step. On subsequent runs of that job, steps that have not changed in the Dockerfile will be reused.
+There is a “DLC set-up” step at the beginning of each job that uses DLC. You are not charged for this step. At the end of each job that uses DLC, the cache upload is done asynchronously, and does not prevent the workflow from continuing to progress. This means that jobs within the same workflow are unlikely to access a cache uploaded from an upstream job. You are not charged for this “DLC teardown” step.
 
-The first run takes over two minutes to build the Docker image. If nothing changes in the Dockerfile before the second run, those steps happen instantly: in zero seconds.
+### Parallelism and DLC
+{: #parallelism-and-dlc }
 
-```yaml
-version: 2.1
-jobs:
-  build:
-    docker:
-      - image: cimg/node:14.17.3
-        auth:
-          username: mydockerhub-user
-          password: $DOCKERHUB_PASSWORD  # context / project UI env-var reference
+DLC operates in the same way for jobs that use parallelism. If a `machine` job using DLC is configured with `parallelism: 2`, two jobs run in parallel. Each virtual machine in this case will have a separate DLC cache, and whichever is saved last will be used for the next build.
 
-    steps:
-      - checkout
-      - setup_remote_docker:
-          docker_layer_caching: true
-      - run: docker build .
-```
+### Scope of cache
+{: #scope-of-cache }
 
-When none of the layers in the image change between job runs, DLC pulls the layers from cache from the image that was built previously and reuses those instead of rebuilding the entire image.
+With DLC enabled, the entirety of `/var/lib/docker` is cached.
```

Comment:
I'm wondering if this is slightly misleading.

It is `/var/lib/docker` that we backup, but we do clean it up first (for example we wont backup volumes created during the job run). We've seen cases where customers have tried to store their own files in `/var/lib/docker`, this is not supported and depending on how they do it can corrupt the cache resulting in us throwing away the whole cache.

Reply:
do we need to include anything about /var/lib/docker? those sound like implementation details that may change in the future?

Reply:
That's sorta my feeling too. If we don't need to say it then we shouldn't.

Reply:
@rosieyohannan i think let's yank it out